components/UserDashboard.tsx

```diff
@@ -38,41 +53,74 @@ const UserDashboardItem: React.FunctionComponent<User> = ({
 
 const UserDashboard: React.FunctionComponent = () => {
   const [users, setUsers] = useState<User[]>();
-
-  const getUsers = async (): Promise<void> => {
-    try {
-      const response = await fetch("http://localhost:3000/api/admin/users", {
-        method: "GET",
-        headers: { "content-type": "application/json" },
-        redirect: "follow",
-      });
-      const data = await response.json();
-      setUsers(data.users);
-    } catch (err) {
-      throw new Error(err.message);
-    }
-  };
+  const [index, setIndex] = useState(0);
+  const [uiPage, setUIPage] = useState(0);
+  const [numUIPages, setNumUIPages] = useState(0);
+  const [pageCache, setPageCache] = useState<Record<number, User[]>>({});
 
   useEffect(() => {
-    getUsers();
-  }, []);
+    const getUsers = async (pageNumber: number): Promise<void> => {
+      try {
+        if (pageNumber in pageCache) {
+          // Page already in cache; no need to make a request!
+          setUsers(pageCache[pageNumber]);
+        } else {
+          const response = await fetch(
+            `http://localhost:3000/api/admin/users?pageNumber=${pageNumber}`,
+            {
+              method: "GET",
+              headers: { "content-type": "application/json" },
+              redirect: "follow",
+            }
+          );
+          const data = await response.json();
+          setUsers(data.users);
+          setNumUIPages(Math.ceil(data.total / UI_PAGE_SIZE));
+          setPageCache({
+            ...pageCache,
+            [pageNumber]: data.users,
+          });
+        }
+      } catch (err) {
+        throw new Error(err.message);
+      }
+    };
+    const updateUIPage = (): void => {
+      const [backendPage, startIndex] = getBackendPageNumber(uiPage);
+      setIndex(startIndex);
+      getUsers(backendPage);
+    };
+    updateUIPage();
+  }, [uiPage, pageCache]);
   return (
     <div>
-      <div className="flex flex-row justify-between text-sm text-center text-unselected tracking-wide">
+      <div className="flex flex-row justify-between text-sm text-center text-unselected tracking-wide mt-10">
         <p>Name</p>
         <p>Email</p>
         <p>Phone</p>
       </div>
       <hr className="border-unselected border-opacity-50" />
       <img src="" alt="" />
-      {users?.slice(0, 7).map((user) => (
+      {users?.slice(index, index + UI_PAGE_SIZE).map((user) => (
         <UserDashboardItem
           name={user.name}
           email={user.email}
           image={user.image}
           phoneNumber={user.phoneNumber}
         />
       ))}
+      <PageNav
+        currentPage={uiPage + 1}
+        numPages={numUIPages}
+        prevPage={() => {
+          setUIPage(uiPage - 1);
+        }}
+        nextPage={() => {
+          setUIPage(uiPage + 1);
+        }}
```

Comment:
Naming suggestion: `onPrevPage` / `onNextPage`, since they're kind of like event handlers for this component.